src/debugger/debug_workflow.ts

```diff
@@ -0,0 +1,156 @@
+
+    const rows = await this.#wfe.userDatabase.queryWithClient<transaction_outputs>(
+      client,
+      query,
+      this.workflowUUID,
+      funcID
+    );
+
+    if (rows.length === 0 || rows.length > 1) {
+      this.logger.error("Unexpected! This should never happen during debug. Returned rows: " + rows.toString());
+      throw new DBOSDebuggerError(`This should never happen during debug. Returned ${rows.length} rows: ` + rows.toString());
+    }
+
+    if (JSON.parse(rows[0].error) != null) {
+      throw deserializeError(JSON.parse(rows[0].error)); // We don't replay errors.
+    }
+
+    const res: RecordedResult<R> = {
+      output: JSON.parse(rows[0].output) as R,
+      txn_snapshot: rows[0].txn_snapshot,
+      txn_id: rows[0].txn_id,
+    }
+
+    // Send a signal to the debug proxy.
+    // TODO: use the real one.
+    await this.#wfe.userDatabase.queryWithClient(client, `--proxy:${res.txn_id}`);
+
+    return res;
+  }
+
+  /**
+   * Execute a transactional function in debug mode.
+   * It connects to a debug proxy and everything should be read-only.
+   */
+  async transaction<T extends any[], R>(txn: Transaction<T, R>, ...args: T): Promise<R> {
+    const config = this.#wfe.transactionConfigMap.get(txn.name);
+    if (config === undefined) {
+      throw new DBOSDebuggerError(`Transaction ${txn.name} not registered!`);
+    }
+    // const readOnly = true; // TODO: eventually, this transaction must be read-only.
+    const funcId = this.functionIDGetIncrement();
+    const span: Span = this.#wfe.tracer.startSpan(txn.name, {}, this.span);
+    let check: RecordedResult<R>;
+
+    const wrappedTransaction = async (client: UserDatabaseClient): Promise<R> => {
+      // Original result must exist during replay.
+      const tCtxt = new TransactionContextImpl(this.#wfe.userDatabase.getName(), client, this,
+      span, this.#wfe.logger, funcId, txn.name);
+      check = await this.checkExecution<R>(client, funcId);
+
+      // Execute the user's transaction.
+      const result = await txn(tCtxt, ...args);
+      return result;
+    };
+
+    const result = await this.#wfe.userDatabase.transaction(wrappedTransaction, config);
+
+    if (JSON.stringify(check!.output) !== JSON.stringify(result)) {
+      this.logger.error(`Detected different transaction output than the original one!\n Expected: ${JSON.stringify(result)}\n Received: ${JSON.stringify(check!.output)}`);
+    }
+    return check!.output; // Always return the recorded result.
+  }
+
+  external<T extends any[], R>(commFn: Communicator<T, R>, ...args: T): Promise<R> {
+    throw new Error("Method not implemented");
+  }
+
+  // Invoke the debugWorkflow() function instead.
+  async childWorkflow<T extends any[], R>(wf: Workflow<T, R>, ...args: T): Promise<WorkflowHandle<R>> {
+    const funcId = this.functionIDGetIncrement();
+    const childUUID: string = this.workflowUUID + "-" + funcId;
```

Comment:
on an unrelated note: it might be a cause of future headaches to have children WF have a non-UUID identifier.

Reply:
We can fix that later.. Currently we don't enforce any UUID format.